planet/clients/orders.py

```diff
@@ -98,21 +98,34 @@ def _stats_url(self):
     def _request(self, url, method, data=None, params=None, json=None):
         return Request(url, method=method, data=data, params=params, json=json)
 
-    async def _do_request(
-        self,
-        request: Request
-    ) -> Response:
+    def _order_factory(self, contents: dict) -> Order:
+        """Convert API response contents to an instance of Order."""
+        links = contents.get("_links", [])
+        results = []
+
+        for res in links.get("results", []):
+            expiration = res.get("expires_at", None)
+            if expiration:
+                expiration = arrow.get(expiration).datetime
```

Comment:
Python's datetime module doesn't parse the API's expiration dates, so I'm bringing in the Arrow project.